src/__tests__/ApolloClient.ts

```diff
@@ -1205,235 +1204,181 @@ describe("ApolloClient", () => {
-                  }, 50);
-                }
+        it("with a replacement of nested array (wf)", async () => {
+          const client = newClient();
+          const observable = client.watchQuery<Data>({ query });
+          const stream = new ObservableStream(observable);
 
-                if (count === 2) {
-                  expect(result.data!.people.friends).toEqual([bestFriend]);
-                  resolve();
+          {
+            const result = await stream.takeNext();
+
+            expect(result.data).toEqual(data);
+            expect(observable.getCurrentResult().data).toEqual(data);
+
+            const bestFriends = result.data!.people.friends.filter(
+              (x) => x.type === "best"
+            );
+
+            // this should re call next
+            client.writeFragment({
+              id: `Person${result.data!.people.id}`,
+              fragment: gql`
+                fragment bestFriends on Person {
+                  friends {
+                    id
+                  }
                 }
+              `,
+              data: {
+                friends: bestFriends,
+                __typename: "Person",
               },
             });
           }
-        );
 
-        itAsync(
-          "with a value change inside a nested array (wf)",
-          (resolve, reject) => {
-            let count = 0;
-            const client = newClient();
-            const observable = client.watchQuery<Data>({ query });
-            observable.subscribe({
-              next: (result) => {
-                count++;
-                if (count === 1) {
-                  expect(result.data).toEqual(data);
-                  expect(observable.getCurrentResult().data).toEqual(data);
-                  const friends = result.data!.people.friends;
-
-                  // this should re call next
-                  client.writeFragment({
-                    id: `Person${result.data!.people.id}`,
-                    fragment: gql`
-                      fragment bestFriends on Person {
-                        friends {
-                          id
-                          type
-                        }
-                      }
-                    `,
-                    data: {
-                      friends: [
-                        { ...friends[0], type: "okayest" },
-                        { ...friends[1], type: "okayest" },
-                      ],
-                      __typename: "Person",
-                    },
-                  });
-
-                  setTimeout(() => {
-                    if (count === 1)
-                      reject(
-                        new Error(
-                          "writeFragment did not re-call observable with next value"
-                        )
-                      );
-                  }, 50);
-                }
+          {
+            const result = await stream.takeNext();
+            expect(result.data!.people.friends).toEqual([bestFriend]);
```

Comment:
```suggestion
            const result = await stream.takeNext({timeout: 50});
```

Do we want to keep that original timeout here - do you know if there was any reason for that specific timeout?

Reply:
I understood this just to way a long enough time to make sure that the `writeFragment` updated the result. I don't think there was much significance to the 50ms specifically. Since `takeNext` is defaulted to 100ms, I figured it was probably ok to let it wait longer.